examples/plot_minimal_pydeseq2_pipeline.py

```diff
@@ -0,0 +1,151 @@
+"""
+PyDESeq 2 pipeline
+===========================
+
+This notebook gives a minimalistic example of how to perform DEA using PyDESeq2.
+
+It allows you to run the PyDESeq2 pipeline either on synthetic data or on one of
+the following TCGA datasets:
+
+- TCGA-BRCA
+- TCGA-COAD
+- TCGA-LUAD
+- TCGA-LUSC
+- TCGA-PAAD
+- TCGA-PRAD
+- TCGA-READ
+- TCGA-SKCM.
+
+While the synthetic data is part of this repository, TCGA data should be
+downloaded as per the instructions in `datasets/`.
+
+Running this pipeline may take a few minutes (~5-10 min for TCGA) depending on
+your setup and on the chosen dataset.
+"""
+
+import os
+import pickle as pkl
+
+from pydeseq2.DeseqDataSet import DeseqDataSet
+from pydeseq2.DeseqStats import DeseqStats
+from pydeseq2.utils import load_example_data
+
+SAVE = False  # whether to save the outputs of this notebook
+
+# %%
+# Data loading
+# ------------
+#
+# See the `datasets` readme for the required data organization.
+#
+
+DATASET = "synthetic"  # or 'TCGA-BRCA', 'TCGA-COAD', etc.
+
+OUTPUT_PATH = f"../output_files/{DATASET}"
+os.makedirs(OUTPUT_PATH, exist_ok=True)  # Create path if it doesn't exist
+
+counts_df = load_example_data(
+    modality="raw_counts",
+    dataset=DATASET,
+    debug=False,
+)
+
+clinical_df = load_example_data(
+    modality="clinical",
+    dataset=DATASET,
+    debug=False,
+)
+
+print(counts_df)
+
+# %%
+# Remove samples for which `high_grade` is NaN.
+
+if DATASET != "synthetic":
+    samples_to_keep = ~clinical_df.high_grade.isna()
+    samples_to_keep.sum()
+    counts_df = counts_df.loc[samples_to_keep]
+    clinical_df = clinical_df.loc[samples_to_keep]
+
+# %%
+# Filter out genes that have less than 10 counts in total
+
+genes_to_keep = counts_df.columns[counts_df.sum(axis=0) >= 10]
+len(genes_to_keep)
+
+counts_df = counts_df[genes_to_keep]
+
+# %%
+# 1 - Read counts modeling with the `DeseqDataSet` class
+# ------------------------------------------------------
+#
+# We start by creating a `DeseqDataSet` object with the count and clinical data.
+# Here, we use 8 threads, feel free to adapt this to your setup or to set
+# to `None` to use all available CPUs.
+#
+# Another option of interest is the `refit_cooks` argument
+# (set to `True` by default), which controls whether Cooks outlier
+# should be refitted – this is advised, in general.
+#
+# .. note::
+#  in the case of the provided synthetic data, there won't be any Cooks
+#  outliers.
+#
+# Start by creating a DeseqDataSet
+dds = DeseqDataSet(
+    counts_df,
+    clinical_df,
+    design_factors="condition" if DATASET == "synthetic" else "high_grade",
```

Comment:
```suggestion
    design_factors="condition",
```

Reply:
again, I think this should be explained, not removed?

Reply:
Not sure, the `high_grade` field is specific to TCGA data. We could add a comment saying that `"condition"` is a column of `clinical_df`